### PR TITLE
HTCONDOR-1004 Miscellaneous pre-release UI/UX improvements.

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
The machine-table stuff could be pulled out into another branch, especially if that helps merge Mat's changes, but the actual effect on the code is small.

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed
- [ ] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_tools/htcondor_cli/annex.py [hunk 1]
```diff
@@ -174,21 +193,44 @@ def __init__(self, logger, **options):
...
+        if the_annex_name is not None:
+            constraint = f'TargetAnnexName == "{the_annex_name}"'
+        target_jobs = schedd.query(
+            constraint,
+            opts=htcondor.QueryOpts.DefaultMyJobsOnly,
+            projection=['JobStatus', 'TargetAnnexName', 'hpc_annex_nodes'],
+        )
         for job in target_jobs:
+            target_annex_name = job['TargetAnnexName']
+            bump(all_jobs, target_annex_name, 1)
```
Comment (jasoncpatton) [inline src/condor_tools/htcondor_cli/annex.py:220]:
```suggestion
            all_jobs[target_annex_name] += 1
```

File: src/condor_tools/htcondor_cli/annex.py [hunk 2]
```diff
@@ -174,21 +193,44 @@ def __init__(self, logger, **options):
...
+
             if job['JobStatus'] == htcondor.JobStatus.RUNNING:
-                count = running_jobs.get(job['TargetAnnexName'], 0)
-                count += 1
-                running_jobs[job['TargetAnnexName']] = count
+                bump(running_jobs, target_annex_name, 1)
```
Comment (jasoncpatton) [inline src/condor_tools/htcondor_cli/annex.py:223]:
```suggestion
                running_jobs[target_annex_name] += 1
```

File: src/condor_tools/htcondor_cli/annex_create.py [hunk 3]
```diff
@@ -22,8 +23,102 @@
...
-# FIXME: Make this come from config? At least some place not hardcoded.
-MACHINE_QUEUE_MAP = {"stampede2": "normal"}
+#
+# For queues with the whole-node allocation policy, cores and RAM -per-node
+# are only informative at the moment; we could compute the necessary number
+# of nodes for the user (as long as we echo it back to them), but let's not
+# do that for now and just tell the user to size their requests with the
+# units appropriate to the queue.
+#
+MACHINE_TABLE = {
```
Comment (jasoncpatton) [inline src/condor_tools/htcondor_cli/annex_create.py:33]: Seems like we should store this in an external JSON file or something, not important now though.

File: src/condor_tools/htcondor_cli/annex_create.py [hunk 4]
```diff
@@ -391,7 +494,8 @@ def annex_create(
     # so even if it's wrong, it will at least consistently so.
     username = getpass.getuser()
 
-    if target not in MACHINE_QUEUE_MAP:
+    target = target.lower()
```
Comment (jasoncpatton) [inline src/condor_tools/htcondor_cli/annex_create.py:497]: Nitpick (and this is what I should have done before), this is what [`str.casefold()`](https://docs.python.org/3.6/library/stdtypes.html#str.casefold) is for:
```suggestion
    target = target.casefold()
```

**Review by jasoncpatton — CHANGES_REQUESTED** [review]
`defaultdict` exists to replace the `bump` function, otherwise this all looks good.

File: src/condor_tools/htcondor_cli/annex_create.py (hunk 3 above)
Comment (Todd-L-Miller) [reply src/condor_tools/htcondor_cli/annex_create.py:33]: Agreed, added to the long list of "other" things we need to to do.

**Comment by Todd-L-Miller** [comment]
Thanks for the `defaultdict` tip.  I'm going to check to make sure the changes work, and then repeat them for `increment()` and friends.

**Review by jasoncpatton — APPROVED** [review]
LGTM!

**Comment by jasoncpatton** [comment]
Waiting for the green from the clang builds...

**Pull request merged**